portability-types-common/src/main/java/org/datatransferproject/types/common/models/media/MediaContainerResource.java

```diff
@@ -0,0 +1,149 @@
+  private final Collection<MediaAlbum> albums;
+
+  @JsonCreator
+  public MediaContainerResource(
+      @JsonProperty("albums") Collection<MediaAlbum> albums,
+      @JsonProperty("photos") Collection<PhotoModel> photos,
+      @JsonProperty("videos") Collection<VideoObject> videos) {
+    this.albums = albums == null ? ImmutableList.of() : albums;
+    this.photos = photos == null ? ImmutableList.of() : photos;
+    this.videos = photos == null ? ImmutableList.of() : videos;
+  }
+
+  public Collection<MediaAlbum> getAlbums() {
+    return albums;
+  }
+
+  public Collection<PhotoModel> getPhotos() {
+    return photos;
+  }
+
+  public Collection<VideoObject> getVideos() {
+    return videos;
+  }
+
+  @Override
+  public Map<String, Integer> getCounts() {
+    return new ImmutableMap.Builder<String, Integer>()
+        .put(ALBUMS_COUNT_DATA_NAME, albums.size())
+        .put(PHOTOS_COUNT_DATA_NAME, photos.size())
+        .put(VIDEOS_COUNT_DATA_NAME, videos.size())
+        .build();
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    MediaContainerResource that = (MediaContainerResource) o;
+    return Objects.equals(getAlbums(), that.getAlbums())
+        && Objects.equals(getPhotos(), that.getPhotos())
+        && Objects.equals(getVideos(), that.getVideos());
+  }
+
+  public void transmogrify(TransmogrificationConfig config) {
+    ensureRootAlbum(config.getAlbumAllowRootPhotos());
+    transmogrifyTitles(config);
+    
+    // TODO(#1000): This splitting code isn't entirely correct since it assumes all the album items
+    //  are present in this resource, when they could be split up it also assumes that all the
+    //  albums that the images correspond to are also in this resource which may not be the case
+    //  so in the meantime, i'm removing this code and it can be added in in the future.
+    //  ensureAlbumSize(config.getAlbumMaxSize());
+  }
+
+  // Coerce the photos of the transfer using the specification provided, e.g.
+  // limiting max title length or removing forbidden characters, etc.
+  private void transmogrifyTitles(TransmogrificationConfig config) {
+    // Replaces forbidden characters and makes sure that the title is not too long
+    for (PhotoModel photo : photos) {
+      photo.cleanTitle(
+          config.getPhotoTitleForbiddenCharacters(),
+          config.getPhotoTitleReplacementCharacter(),
+          config.getPhotoTitleMaxLength());
+    }
+
+    for (VideoObject video : videos) {
+      video.cleanName(
+          config.getPhotoTitleForbiddenCharacters(),
+          config.getPhotoTitleReplacementCharacter(),
+          config.getPhotoTitleMaxLength());
+    }
+
+    for (MediaAlbum album : albums) {
+      album.cleanName(
+          config.getPhotoTitleForbiddenCharacters(),
+          config.getPhotoTitleReplacementCharacter(),
+          config.getPhotoTitleMaxLength());
+    }
```

Comment:
seems like we are using the photos-related config for videos and albums as well. Is that intentional?

Reply:
Good catch, done!